server/src/main/scala/io/delta/standalone/internal/DeltaSharedTableLoader.scala

```diff
@@ -68,20 +70,46 @@ class DeltaSharedTable(
     evaluatePredicateHints: Boolean) {
 
   private val conf = withClassLoader {
-    new Configuration()
+    val defaultConfig = new Configuration()
+    val azureStorageAccount = sys.env.getOrElse("AZURE_STORAGE_ACCOUNT", "")
+    val azureStorageKey = sys.env.getOrElse("AZURE_STORAGE_KEY", "")
+    // Set access keys for Azure Data Lake Storage Gen1 and Gen2
+    defaultConfig.set(s"fs.azure.account.key.$azureStorageAccount.blob.core.windows.net",
+      azureStorageKey)
+    defaultConfig.set(s"fs.azure.account.key.$azureStorageAccount.dfs.core.windows.net",
+      azureStorageKey)
+    defaultConfig
   }
 
   private val deltaLog = withClassLoader {
     val tablePath = new Path(tableConfig.getLocation)
     val fs = tablePath.getFileSystem(conf)
-    if (!fs.isInstanceOf[S3AFileSystem]) {
-      throw new IllegalStateException("Cannot share tables on non S3 file systems")
+    if (!(fs.isInstanceOf[S3AFileSystem] ||
+      fs.isInstanceOf[NativeAzureFileSystem] ||
+      fs.isInstanceOf[AzureBlobFileSystem])) {
+      throw new IllegalStateException("Cannot share tables on non S3 or non Azure file systems")
```

Comment:
Could this error message be a bit more scalable?
Something like `F" Filesystem {fs.type} is unsupported"`

Reply:
Thinking ahead as this scales to multiple clouds, even on-prem object stores... should cloud specific implementations reside in their own class files.
Can one instance of the server support multiple both Azure and AWS file systems?